tree/treeplayer/inc/ROOT/TDFInterface.hxx

```diff
@@ -364,6 +364,27 @@ public:
       TInterface<TTraits::TakeFirstParameter_t<decltype(upcastNode)>> upcastInterface(fProxiedPtr, fImplWeakPtr);
       // build a string equivalent to
       // "(TInterface<nodetype*>*)(this)->Snapshot<Ts...>(treename,filename,*(ColumnNames_t*)(&columnList))"
+
+      // The interpretation has multiple steps including
+      // 1. Parse
+      // 2. Codegen
+      // 3. Link
+      // 4. run static initializers
+      // 5. run user code
+      // 6. run static destructors.
+      //
+      // Where 1,2,3 explicitly modify the state of the (global) interpreter and must be (and, of course, are)
+      // covered by the ROOT global lock.  Technically, cling should release the lock when executing user
```

Comment:
If we don't find a find for cling to trigger (likely indirectly via a callback) the release of the lock then we have a fundamental problem (guaranteed deadlock).